[JSC] Remove typedArrayStorageType

Constellation opened the pull request:
#### e98e128329464e856b7fb19d157b4432a9eb69f3
<pre>
[JSC] Remove typedArrayStorageType
<a href="https://bugs.webkit.org/show_bug.cgi?id=243147">https://bugs.webkit.org/show_bug.cgi?id=243147</a>

Reviewed by Alexey Shvayka.

Sampling profiler showed that dispatching typed-array&apos;s derived types in typed array method is costly.
And we are using typedArrayStorageType from ClassInfo from Structure from object. But this is not necessary at all
since JSType of object can represent the necessary information completely.
This patch removes typedArrayStorageType from ClassInfo. And we get the same information from JSType of the object.

* Source/JavaScriptCore/API/JSTypedArray.cpp:
(toJSTypedArrayType):
(JSValueGetTypedArrayType):
(JSObjectGetTypedArrayByteLength):
* Source/JavaScriptCore/API/glib/JSCValue.cpp:
(jsc_value_typed_array_get_type):
* Source/JavaScriptCore/bytecode/GetByVariant.cpp:
(JSC::GetByVariant::canMergeIntrinsicStructures const):
* Source/JavaScriptCore/bytecode/Repatch.cpp:
(JSC::tryCacheArrayGetByVal):
(JSC::tryCacheArrayPutByVal):
* Source/JavaScriptCore/bytecode/SpeculatedType.cpp:
(JSC::speculationFromClassInfoInheritance):
* Source/JavaScriptCore/dfg/DFGAbstractInterpreterInlines.h:
(JSC::DFG::AbstractInterpreter&lt;AbstractStateType&gt;::executeEffects):
* Source/JavaScriptCore/dfg/DFGByteCodeParser.cpp:
(JSC::DFG::ByteCodeParser::handleIntrinsicGetter):
* Source/JavaScriptCore/jit/IntrinsicEmitter.cpp:
(JSC::IntrinsicGetterAccessCase::canEmitIntrinsicGetter):
(JSC::IntrinsicGetterAccessCase::emitIntrinsicGetter):
* Source/JavaScriptCore/runtime/ClassInfo.h:
* Source/JavaScriptCore/runtime/JSArrayBufferView.cpp:
(JSC::validateTypedArray):
(JSC::elementSize): Deleted.
* Source/JavaScriptCore/runtime/JSCast.h:
* Source/JavaScriptCore/runtime/JSGenericTypedArrayViewConstructorInlines.h:
(JSC::constructGenericTypedArrayViewWithArguments):
* Source/JavaScriptCore/runtime/JSGenericTypedArrayViewInlines.h:
(JSC::JSGenericTypedArrayView&lt;Adaptor&gt;::set):
* Source/JavaScriptCore/runtime/JSGenericTypedArrayViewPrototypeFunctions.h:
(JSC::genericTypedArrayViewProtoFuncSet):
(JSC::genericTypedArrayViewProtoFuncSlice):
(JSC::genericTypedArrayViewPrivateFuncSubarrayCreate):
* Source/JavaScriptCore/runtime/JSGlobalObject.h:
(JSC::JSGlobalObject::isOriginalTypedArrayStructure):
* Source/JavaScriptCore/runtime/JSTypedArrayViewPrototype.cpp:
(JSC::JSC_DEFINE_HOST_FUNCTION):
* Source/JavaScriptCore/runtime/Structure.h:
(JSC::Structure::hijacksIndexingHeader const):
* Source/JavaScriptCore/runtime/StructureInlines.h:
(JSC::Structure::hasIndexingHeader const):
* Source/JavaScriptCore/runtime/TypedArrayType.h:
(JSC::typedArrayType):
(JSC::isTypedView):
(JSC::elementSize):
(JSC::contentType):

Canonical link: <a href="https://commits.webkit.org/252772@main">https://commits.webkit.org/252772@main</a>
</pre>

shvaikalesh reviewed (approved):
That's a nice simplification!

Constellation committed:
[JSC] Remove typedArrayStorageType
https://bugs.webkit.org/show_bug.cgi?id=243147

Reviewed by Alexey Shvayka.

Sampling profiler showed that dispatching typed-array's derived types in typed array method is costly.
And we are using typedArrayStorageType from ClassInfo from Structure from object. But this is not necessary at all
since JSType of object can represent the necessary information completely.
This patch removes typedArrayStorageType from ClassInfo. And we get the same information from JSType of the object.

* Source/JavaScriptCore/API/JSTypedArray.cpp:
(toJSTypedArrayType):
(JSValueGetTypedArrayType):
(JSObjectGetTypedArrayByteLength):
* Source/JavaScriptCore/API/glib/JSCValue.cpp:
(jsc_value_typed_array_get_type):
* Source/JavaScriptCore/bytecode/GetByVariant.cpp:
(JSC::GetByVariant::canMergeIntrinsicStructures const):
* Source/JavaScriptCore/bytecode/Repatch.cpp:
(JSC::tryCacheArrayGetByVal):
(JSC::tryCacheArrayPutByVal):
* Source/JavaScriptCore/bytecode/SpeculatedType.cpp:
(JSC::speculationFromClassInfoInheritance):
* Source/JavaScriptCore/dfg/DFGAbstractInterpreterInlines.h:
(JSC::DFG::AbstractInterpreter<AbstractStateType>::executeEffects):
* Source/JavaScriptCore/dfg/DFGByteCodeParser.cpp:
(JSC::DFG::ByteCodeParser::handleIntrinsicGetter):
* Source/JavaScriptCore/jit/IntrinsicEmitter.cpp:
(JSC::IntrinsicGetterAccessCase::canEmitIntrinsicGetter):
(JSC::IntrinsicGetterAccessCase::emitIntrinsicGetter):
* Source/JavaScriptCore/runtime/ClassInfo.h:
* Source/JavaScriptCore/runtime/JSArrayBufferView.cpp:
(JSC::validateTypedArray):
(JSC::elementSize): Deleted.
* Source/JavaScriptCore/runtime/JSCast.h:
* Source/JavaScriptCore/runtime/JSGenericTypedArrayViewConstructorInlines.h:
(JSC::constructGenericTypedArrayViewWithArguments):
* Source/JavaScriptCore/runtime/JSGenericTypedArrayViewInlines.h:
(JSC::JSGenericTypedArrayView<Adaptor>::set):
* Source/JavaScriptCore/runtime/JSGenericTypedArrayViewPrototypeFunctions.h:
(JSC::genericTypedArrayViewProtoFuncSet):
(JSC::genericTypedArrayViewProtoFuncSlice):
(JSC::genericTypedArrayViewPrivateFuncSubarrayCreate):
* Source/JavaScriptCore/runtime/JSGlobalObject.h:
(JSC::JSGlobalObject::isOriginalTypedArrayStructure):
* Source/JavaScriptCore/runtime/JSTypedArrayViewPrototype.cpp:
(JSC::JSC_DEFINE_HOST_FUNCTION):
* Source/JavaScriptCore/runtime/Structure.h:
(JSC::Structure::hijacksIndexingHeader const):
* Source/JavaScriptCore/runtime/StructureInlines.h:
(JSC::Structure::hasIndexingHeader const):
* Source/JavaScriptCore/runtime/TypedArrayType.h:
(JSC::typedArrayType):
(JSC::isTypedView):
(JSC::elementSize):
(JSC::contentType):

Canonical link: https://commits.webkit.org/252772@main
